Assets/Scripts/Battle/ProgressBarController.cs

```diff
@@ -0,0 +1,82 @@
+﻿using UnityEngine;
+using UnityEngine.UI;
+using System.Collections;
+
+[RequireComponent(typeof(Image))]
+public class ProgressBarController : MonoBehaviour
+{
+    [SerializeField]
+    Image gaugeImage;
+
+    float minValue = 0;
+    float maxValue = 100;
+    float currentValue;
+    float newValue;
+    bool isInitialized = false;
+    [SerializeField]
+    float tick;
+
+    public void Init(int initValue)
+    {
+        if (isInitialized)
+        {
+            StartCoroutine(SetAnimation(initValue));
+            return;
+        }
+
+        this.currentValue = initValue;
+        this.maxValue = initValue;
+        isInitialized = true;
+    }
+
+    public void AddOne()
```

Comment:
Nie ogarniam trochę tego :)?
AddOne dodaje 5, niżej jest metoda która wygląda analogicznie i jak wywołamy Add(1) to będzie inne zachowanie niż w AddOne() mimo że nazwa sugeruje to samo. Może użyć domyślnego parametru Add(int i =  1) ?